dotnet/src/Agents/OpenAI/OpenAIAssistantConfiguration.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Net.Http;
+using Azure.AI.OpenAI.Assistants;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI;
+
+/// <summary>
+/// Configuration to target an OpenAI Assistant API.
+/// </summary>
+public sealed class OpenAIAssistantConfiguration
+{
+    /// <summary>
+    /// The Assistants API Key.
+    /// </summary>
+    public string ApiKey { get; }
+
+    /// <summary>
+    /// An optional endpoint if targeting Azure OpenAI Assistants API.
+    /// </summary>
+    public string? Endpoint { get; }
+
+    /// <summary>
+    /// An optional API version override.
+    /// </summary>
+    public AssistantsClientOptions.ServiceVersion? Version { get; init; }
+
+    /// <summary>
+    /// Custom <see cref="HttpClient"/> for HTTP requests.
+    /// </summary>
+    public HttpClient? HttpClient { get; init; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenAIAssistantConfiguration"/> class.
+    /// </summary>
+    /// <param name="apiKey">The Assistants API Key</param>
+    /// <param name="endpoint">An optional endpoint if targeting Azure OpenAI Assistants API</param>
+    public OpenAIAssistantConfiguration(string apiKey, string? endpoint = null)
+    {
+        Verify.NotNullOrWhiteSpace(apiKey);
+        if (!string.IsNullOrWhiteSpace(endpoint))
+        {
+            Verify.StartsWith(endpoint!, "https://", "The Azure OpenAI endpoint must start with 'https://'");
+        }
```

Comment:
No need to check for null or whitespace since `Verify.StartsWith` already does this:
https://github.com/microsoft/semantic-kernel/blob/62cc40d063708befd0a6a8062676bfd12c6ca65b/dotnet/src/InternalUtilities/src/Diagnostics/Verify.cs#L110-L114

```suggestion
        Verify.StartsWith(endpoint!, "https://", "The Azure OpenAI endpoint must start with 'https://'");
```

We should probably add `[NotNull]` attribute to first parameter of `StartsWith` method to avoid suppressing possible null reference.

Reply:
FWIW - Debug.Assert isn't a runtime check.

Reply:
Ok, refreshed my understanding of what's going on here.  Since `endpoint` is only provided for Azure OpenAI, it only need to be validated when provided.  I've added comment and actioned on the rest of your input.